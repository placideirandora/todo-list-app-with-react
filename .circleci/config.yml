version: 2.1
jobs:
  build:
    docker:
      - image: circleci/node:10.16.0

    working_directory: ~/todo-list-app-with-react-js

    steps:
      - checkout
      - restore_cache:
          keys:
            - v1-dependencies-{{ checksum "package.json" }}
            - v1-dependencies-

      - run: npm install

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}

      - run: npm run test:coverage
