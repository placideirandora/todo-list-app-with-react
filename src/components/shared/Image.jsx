import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const Image = ({ source, width, height }) => {
  return (
    <Fragment>
      <img src={source} alt="image" width={width} height={height} />
    </Fragment>
  );
};

Image.protoTypes = {
  source: PropTypes.string.isRequired,
  width: PropTypes.number,
  height: PropTypes.number
};

Image.defaultProps = {
  width: 400,
  height: 300
};

export default Image;
