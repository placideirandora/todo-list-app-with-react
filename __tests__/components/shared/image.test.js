import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as adapter from '../../config/enzyme.config';
import Image from '../../../src/components/shared/Image';

const props = {
  source: 'image location'
};

describe('Footer Component Test Cases', () => {
  it('should render the component without any failure', () => {
    const component = shallow(<Image {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
