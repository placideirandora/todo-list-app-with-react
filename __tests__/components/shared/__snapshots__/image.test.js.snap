// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Test Cases should render the component without any failure 1`] = `
<Fragment>
  <img
    alt="image"
    height={300}
    src="image location"
    width={400}
  />
</Fragment>
`;
